examples/camera-app/linux/src/clusters/camera-avstream-mgmt/camera-av-stream-manager.cpp

```diff
@@ -278,6 +278,17 @@ Protocols::InteractionModel::Status CameraAVStreamManager::SnapshotStreamAllocat
             {
                 stream.isAllocated = true;
 
+                // Set the optional Watermark and OSD values if provided in the allocation
```

Comment:
If they are not provided... is it expected that allocateArgs already has some other values for them and those other values need to be preserved?  If so, that's the thing worth documenting, because it explains why this logic is as it is.

And if that's not expected, then these sets can probably be unconditional.